### PR TITLE
Taylor Keller

### PR DESCRIPTION
**Pull request by takeller**
Was sooooo close to getting the sell method to work. I think I would have had it with a few more minutes.

### CODE REVIEW

File: lib/event.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    food_trucks = @food_trucks.map { |food_truck| food_truck.name }
+  end
+
+  def food_trucks_that_sell(item)
+    food_trucks_with_item = @food_trucks.find_all do |food_truck|
+      food_truck.check_stock(item) > 0
+    end
+  end
+
+  def items_in_event
```
Comment (memcmahon) [inline lib/event.rb:24]: Love these helper methods!

File: lib/event.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  end
+
+  def total_inventory
+    total_inventory = items_in_event
+    total_inventory.map do |item, item_info|
+      item_info[:quantity] = total_quantity(item)
+      item_info[:food_trucks] = food_trucks_that_sell(item)
+    end
+    total_inventory.to_h
+  end
```
Comment (memcmahon) [inline lib/event.rb:49]: Great job with this method - though you do not _need_ to assign a variable for `total_inventory`.  You could consolidate like this:

```ruby
items_in_event.map do |item, item_info|
      item_info[:quantity] = total_quantity(item)
      item_info[:food_trucks] = food_trucks_that_sell(item)
end.to_h
```

File: lib/event.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  def sorted_item_list
+    item_list = items_in_event.keys
```
Comment (memcmahon) [inline lib/event.rb:52]: Again, you don't need to assign this variable, you could call `.map` directly on `item_in_event.keys`

File: lib/event.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        food_truck.inventory[item] - quantity_remaining
+      else 
+        quantity_remaining = quantity - food_truck.inventory[item]
+        food_truck.inventory[item] = 0
+      end
+
+    end
+    true
+
+  end
```
Comment (memcmahon) [inline lib/event.rb:84]: You are sooooo close! Definitely on the right track - keep in mind that the resulting quantity for the inventory will not always be 0 but will always be reduced.

File: test/event_test.rb [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    assert_equal expected, @event.sorted_item_list
+  end
+
+  def test_overstocked_items
+    @event.add_food_truck(@food_truck1)
+    @event.add_food_truck(@food_truck2)
+    @event.add_food_truck(@food_truck3)
+    assert_equal [@item1], @event.overstocked_items
+  end
+
```
Comment (memcmahon) [inline test/event_test.rb:132]: Looks like you're missing the `date` method here!

File: test/item_test.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  def test_it_exists
+    assert_instance_of Item, @item1
+    assert_instance_of Item, @item2
+  end
+
+  def test_it_has_readable_attributes
+    assert_equal 'Peach Pie (Slice)', @item1.name
+    assert_equal 'Apple Pie (Slice)', @item2.name
+    assert_equal "$3.75", @item1.price
```
Comment (memcmahon) [inline test/item_test.rb:20]: Keep a close eye on the interaction pattern - `price` should return a float, not a string!

File: lib/event.rb (hunk 2 above)
Comment (takeller) [reply lib/event.rb:49]: This makes sense. I haven't become super comfortable with tagging methods on an end like this, but I'll give it a try.

**Pull request closed without merging**